### PR TITLE
feat(callout-update): update callout component UI

### PR DESCRIPTION
**Pull request by zahnster**
* Updates visual appearance
* Cleans up variant naming

### CODE REVIEW

**Review by shleewhite — APPROVED** [review]
This looks good! I remember there was talk that this component would need to support both React children and markdown. Is that still needed?

**Review by loreina — COMMENTED** [review]
Yay, this is looking so awesome!!!! Just a few lil notes:

- The `Callout` on the Object Details patterns page has text that isn't using `color-text-neutral`, but the other `Callout` using the info variant have that text color
image
- Could we center align the icon when there's no title? It looks a bit higher atm than when there's a title
image
- Would we want to add stories for Callout without a title? Is that a common use case?

**Comment by katieporter** [comment]
This is looking amazing! Loreina covered nearly everything, I just noticed a couple more small things:
- When there is more than one paragraph in a callout, none of the paragraphs have a bottom margin. It's perfect that the last paragraph doesn't have a bottom margin, but ideally any other paragraphs would maintain the default margin.
image
- The callouts on the /components, /primitives, and /layout pages also use `color-text` instead of `color-text-neutral`. They also have extra margin at the bottom because of the built in margin on List. Not sure if we can fix that in the Callout component, but if we can, it'd be great to remove that margin.
- There are also callouts on /patterns/notifications-and-feedback and /patterns/object-details that have the same problem with the text color. Seems like this happens when using the Paragraph or List components instead of using Text. Not sure if there's a great solution for that or if we just need to clean up those instances to use Text instead?


I do have a tiny request and if it's out of scope, that is totally fine. There are some copy tweaks I'd like to make to callouts to make them feel a little more polished:
- We have some callouts titled "Hot accessibility tip". That's, erm, not super on brand lol. Could we update those to just say "Accessibility insight"?
- We have some callouts that use kind of 🚨aggressive emojis🚨 in the title. Could we run a find and replace on the 🚨 emoji and delete all instances of it? It seems extra aggressive now that we've added the warning icon to the warning callout. We've also got one that uses the 🦮  emoji which is cute but looks a little funny now.

**Comment by nkrantz** [comment]
I know it technically passes contrast ratio... but do we _like_ it? It's so blue and kind of hurts my eyes but that could totally just be me.
Screen Shot 2021-11-16 at 1 00 58 PM

File: packages/paste-website/src/pages/components/combobox/index.mdx [hunk 1]
```diff
@@ -181,7 +181,7 @@ In the example below the value of the Combobox is stored in a piece of our appli
 #### useCombobox state hook
 
 <Callout variant="warning">
-  <CalloutTitle>🚨Power user move!🚨</CalloutTitle>
+  <CalloutTitle>Power user move!</CalloutTitle>
```
Comment (shleewhite) [inline packages/paste-website/src/pages/components/combobox/index.mdx:184]: The heading levels are incorrect here, should be an H5

File: packages/paste-website/src/pages/components/index.mdx [hunk 2]
```diff
@@ -41,27 +39,27 @@ import {Callout, CalloutTitle, CalloutText} from '../../components/callout';
 
 <Callout>
   <CalloutTitle as="h4">Don't see a component, primitive, or layout you need listed here?</CalloutTitle>
```
Comment (shleewhite) [inline packages/paste-website/src/pages/components/index.mdx:41]: Should be H2

File: packages/paste-website/src/pages/layout/index.mdx [hunk 3]
```diff
@@ -37,27 +36,27 @@ import {Callout, CalloutTitle, CalloutText} from '../../components/callout';
 
 <Callout>
```
Comment (shleewhite) [inline packages/paste-website/src/pages/layout/index.mdx:37]: Should be H2

File: packages/paste-website/src/pages/patterns/button-vs-anchor/index.mdx [hunk 4]
```diff
@@ -193,7 +193,7 @@ There are some cases where a secondary button may be used for an irreversible ac
 ### Link-style button (tertiary)
 
 <Callout variant="warning">
-  <CalloutTitle>🚨 This is not a variant we recommend you use often.</CalloutTitle>
+  <CalloutTitle>This is not a variant we recommend you use often.</CalloutTitle>
```
Comment (shleewhite) [inline packages/paste-website/src/pages/patterns/button-vs-anchor/index.mdx:196]: Should be H4

File: packages/paste-website/src/pages/patterns/button-vs-anchor/index.mdx [hunk 5]
```diff
@@ -220,7 +220,7 @@ When making this decision, it's **a great opportunity to test** which buttons yo
 ### Buttons with link functionality
 
 <Callout variant="warning">
-  <CalloutTitle>🚨 This is not a variant we recommend you use often.</CalloutTitle>
+  <CalloutTitle>This is not a variant we recommend you use often.</CalloutTitle>
```
Comment (shleewhite) [inline packages/paste-website/src/pages/patterns/button-vs-anchor/index.mdx:223]: Should be H4

File: packages/paste-website/src/pages/patterns/object-details/index.mdx [hunk 6]
```diff
@@ -167,27 +164,25 @@ The most common way to structure an object details page is to first show an over
 <Callout>
   <CalloutTitle>Composing a Description List</CalloutTitle>
```
Comment (shleewhite) [inline packages/paste-website/src/pages/patterns/object-details/index.mdx:165]: Should be H4

File: packages/paste-website/src/pages/patterns/object-details/index.mdx [hunk 7]
```diff
@@ -238,7 +233,7 @@ When you find that customers want to see multiple, top-level content sections, s
 
 Many object detail pages also include a set of actions that can be performed on that object, such as editing or deleting.
 
-<Callout variant="info">
+<Callout>
   <CalloutTitle>This pattern is coming soon!</CalloutTitle>
```
Comment (shleewhite) [inline packages/paste-website/src/pages/patterns/object-details/index.mdx:237]: Should be H4

File: packages/paste-website/src/pages/primitives/index.mdx [hunk 8]
```diff
@@ -40,27 +39,27 @@ import {Callout, CalloutTitle, CalloutText} from '../../components/callout';
 
 <Callout>
   <CalloutTitle as="h4">Don't see a primitive, layout, or component you need listed here?</CalloutTitle>
```
Comment (shleewhite) [inline packages/paste-website/src/pages/primitives/index.mdx:41]: Should be H2

**Review by shleewhite — CHANGES_REQUESTED** [review]
Sorry for the late review - realized there were some incorrect heading levels.

**Comment by zahnster** [comment]
> I know it technically passes contrast ratio... but do we _like_ it? 

Ehh, but I figure that is part of the dark mode color effort in general, there are theming updates on deck there.

**Review by shleewhite — APPROVED** [review]
🚢

**Comment by katieporter** [comment]
This is pretty minor, so if it's a pain to fix, it's not a huge deal. I noticed that any callout with a UL in it is now very indented. Is it possible to get those to match the indentation of the List component? If not, totally fine.
image

I found one more Callout that still has `color-text` on the Notification and feedback pattern page. I assume it's due to how that individual callout was composed?
image

It looks like all paragraph text within a callout has a margin bottom now, which makes the spacing around the callout look a touch uneven. Is it possible to remove the margin bottom from the _last_ element in any callout?
image

**Review by SiTaggart — APPROVED** [review]
This feels like a much nicer API 🎉

**Pull request merged**